tensorflow/python/keras/layers/recurrent_v2.py

```diff
@@ -583,16 +584,18 @@ def cudnn_gru(inputs, init_h, kernel, recurrent_kernel, bias, mask, time_major,
   # Note that the bias was initialized as shape (2, 3 * units), flat it into
   # (6 * units)
   bias = array_ops.split(K.flatten(bias), 6)
-  # Note that the gate order for CuDNN is different from the canonical format.
-  # canonical format is [z, r, h], whereas CuDNN is [r, z, h]. The swap need to
-  # be done for kernel, recurrent_kernel, input_bias, recurrent_bias.
-  # z is update gate weights.
-  # r is reset gate weights.
-  # h is output gate weights.
-  weights[0], weights[1] = weights[1], weights[0]
-  weights[3], weights[4] = weights[4], weights[3]
-  bias[0], bias[1] = bias[1], bias[0]
-  bias[3], bias[4] = bias[4], bias[3]
+
+  if build_info.is_cuda_build:
```

Comment:
This function is already expecting cudnn build, so we probably shouldn't add this check here, and instead, add this at the caller of this function.